src/common/queries/useAccountBalance.ts

```diff
@@ -1,56 +1,45 @@
-import {
-  UseQueryOptions,
-  UseSuspenseQueryOptions,
-  useQuery,
-  useSuspenseQuery,
-} from '@tanstack/react-query';
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
 
-import { AddressBalanceResponse } from '@stacks/stacks-blockchain-api-types';
-
-import { useApi } from '../api/useApi';
+import { getErrorMessage } from '../../api/getErrorMessage';
+import { useApiClient } from '../../api/useApiClient';
 import { ONE_MINUTE } from './query-stale-time';
 
 const ACCOUNT_BALANCE_QUERY_KEY = 'accountBalance';
 
-export function useAccountBalance(
-  address?: string,
-  options: Omit<UseQueryOptions<any, any, AddressBalanceResponse, any>, 'queryKey' | 'queryFn'> = {}
-) {
-  const api = useApi();
+export function useAccountBalance(address?: string) {
+  const apiClient = useApiClient();
   return useQuery({
     queryKey: [ACCOUNT_BALANCE_QUERY_KEY, address],
-    queryFn: () => {
+    queryFn: async () => {
       if (!address) return undefined;
```

Comment:
not needed because of enabled prop

Reply:
Unfortunately `enabled` option in react query only prevents the query from running but does not perform type narrowing